GitCommands/Git/GitTagController.cs

```diff
@@ -0,0 +1,111 @@
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Text;
+
+namespace GitCommands.Git
+{
+    public interface IGitTagController
+    {
+        /// <summary>
+        /// Create the Tag depending on input parameter.
+        /// </summary>
+        /// <param name="revision">Commit revision to be tagged</param>
+        /// <param name="tagName">Name of tag</param>
+        /// <param name="force">Force parameter</param>
+        /// <param name="operationType">Available tag operationType (Lightweight, Annotate, Sign with defaul key, Sign with specific key)</param>
+        /// <param name="tagMessage">Tag Message</param>
+        /// <param name="keyId">Specific Key ID to be used instead of default one</param>
+        /// <returns>Output string from RunGitCmd.</returns>
+        string CreateTag(string revision, string tagName, bool force, GitTagController.TagOperation operationType = GitTagController.TagOperation.Lightweight, string tagMessage = "", string keyId = "");
+    }
+
+
+    public class GitTagController : IGitTagController
+    {
+        private IGitModule _module;
+
+
+        public enum TagOperation
+        {
+            Lightweight = 0,
+            Annotate = 1,
+            SignWithDefaultKey = 2,
+            SignWithSpecifiKey = 3
+        };
+
+
+        public GitTagController(IGitModule module)
+        {
+            _module = module ?? throw new ArgumentNullException("module");
+        }
+
+        /// <summary>
+        /// Create the Tag depending on input parameter.
+        /// </summary>
+        /// <param name="revision">Commit revision to be tagged</param>
+        /// <param name="tagName">Name of tag</param>
+        /// <param name="force">Force parameter</param>
+        /// <param name="operation">Available tag operationType (Lightweight, Annotate, Sign with defaul key, Sign with specific key)</param>
+        /// <param name="tagMessage">Tag Message</param>
+        /// <param name="keyId">Specific Key ID to be used instead of default one</param>
+        /// <returns>Output string from RunGitCmd.</returns>
+        public string CreateTag(string revision, string tagName, bool force, TagOperation operationType = TagOperation.Lightweight, string tagMessage = "", string keyId = "")
+        {
+            if (string.IsNullOrEmpty(revision))
+            {
+                throw new ArgumentNullException("revision");
+            }
+
+            if (string.IsNullOrEmpty(tagName))
+            {
+                throw new ArgumentNullException("tagName");
+            }
+
+            if (operationType > TagOperation.Lightweight)
+            {
+                File.WriteAllText(Path.Combine(_module.GetGitDirectory(), "TAGMESSAGE"), tagMessage);
+            }
+
+            string strCommand = "";
+            string _tagName = tagName.Trim();
+            string _forced = force ? " - f" : "";
+            string _directory = _module.GetGitDirectory();
```

Comment:
please use camelCase notation for local variables and the underscore notation for class instance fields (this is contrary to MS guidelines but made popular by JetBrains Resharper).